src/OSPSuite.R/Services/ContainerTask.cs

```diff
@@ -186,6 +196,25 @@ public void SetValueByPath(IModelCoreSimulation simulation, string path, double
          _logger.AddWarning(Error.CouldNotFindQuantityWithPath(path));
       }
 
+      public double GetValueByPath(IModelCoreSimulation simulation, string path, bool throwIfNotFound)
+      {
+         if (path.Contains(WILD_CARD))
+            throw new OSPSuiteException(Error.CannotGetValueByPathUsingWildCard(path));
+
+         var pathArray = path.ToPathArray();
+         var quantity = simulation.Model.Root.EntityAt<IQuantity>(pathArray);
```

Comment:
you can pass `path` directly, `EntityAt ` can handle it and will split it in case it contains more than one path entry
https://github.com/Open-Systems-Pharmacology/OSPSuite.Core/blob/1ce5c0b852f897b6c5b9aeae19ca3ad9abf9fb7e/src/OSPSuite.Core/Domain/ContainerExtensions.cs#L127-L131

Reply:
and I would look at the SetValueByPath and so the same. Use the method instead of having the same code, three time in the same file